internal/server/events.go

```diff
@@ -61,6 +73,61 @@ func (s *Server) eventsHandler() http.HandlerFunc {
 	}
 }
 
+// eventHandler returns a handler to get a specific event.
+func (s *Server) eventHandler() http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		// Get new event data from TBA if event data is over 24 hours old
+		err := s.updateEvents()
+		if err != nil {
+			ihttp.Error(w, http.StatusInternalServerError)
+			s.logger.Printf("Error: updating event data: %v\n", err)
+			return
+		}
+
+		vars := mux.Vars(r)
```

Comment:
Since you're just getting one var, this can be simplified to `mux.Vars(r)["eventKey"]`